packages/compass-schema/src/stores/store.js

```diff
@@ -217,6 +222,62 @@ const configureStore = (options = {}) => {
       }
     },
 
+    _calculateDepthByPath(input, paths) {
+      if (!input) {
+        return paths;
+      }
+      input.forEach(({types, fields, path, bsonType}) => {
+        const increment = bsonType === 'Array' ? 1 : 0;
+        const score = path.split('.').length + increment;
+        if (!paths[path] || paths[path] < score) {
+          paths[path] = score;
+        }
+        this._calculateDepthByPath(types ?? fields, paths);
+      });
+
+      return paths;
+    },
+
+    calculateSchemaDepth(schema) {
+      const response = this._calculateDepthByPath(schema.fields, {});
+      const values = Object.values(response);
+      return Math.max(...values, ...[0]);
+    },
+
+    _containsGeoData(input) {
+      let result = false;
+      if (!input) {
+        return result;
+      }
+      for (let i = 0; i < input.length; i++) {
+        const { path, values, types, fields } = input[i];
```

Comment:
This can be simplified a bit, I think:

```suggestion
      for (const { path, values, types, fields } of input) {
```